### ISSUE TITLE
Duplicated cards are not presented in the vault

### ISSUE THREAD

**Issue by pawelchmara** [braintree/braintree-web-drop-in#564]
### General information

* SDK/Library version: 1.19.0
* Environment: Sandbox
* Browser and OS: Does not matter

### Issue description

After adding a card that already exists in the vault, duplicated cards are presented only once (tested for card VISA 4111 1111 1111 1111):

![2 card](https://user-images.githubusercontent.com/20394260/66021878-65a15f00-e4ec-11e9-8393-4198247f5bec.png)

After refreshing the page, only one credit card is presented:

![1 card](https://user-images.githubusercontent.com/20394260/66021905-82d62d80-e4ec-11e9-83a0-76bb2618e8ad.png)

What's more, when one card has been deleted will a second card appear.

**Expected behaviour**: all duplicated cards should be presented.

**Comment by hollabaq86** [comment]
👋 @pawelchmara, thanks for reaching out. This is expected behavior - when presenting vaulted cards in the Drop-In, we exclude expired cards and de-dupe duplicate cards by presenting the card with an expiration date that is the farthest in the future.

We can add an option for folks to opt-out of this behavior. We don't have an ETA on when this param would be added, but we'll post updates in this issue and leave this open to track interest.

**Comment by pawelchmara** [comment]
Thank you for your answer and help.

I understand that this is the expected behaviour, but it is inconsistent. Immediately after adding the second card, they should not be presented separately (as in the attached first screenshot). What's more, deleting one card (because only one is presented after a page refresh) should delete all duplicated cards, but it doesn't. This is misleading for the user because after deleting one card, he will see the duplicated card and think that it has not been deleted...

I think that after fixing the above bugs and adding a feature proposed by you, the Dropin will be fully useful.

**Comment by sbosell** [comment]
We are also still seeing this behavior.  If you add a duplicate card the drop in component immediately shows both. When you refresh the page it only shows you one the most recently added.   This component behavior is not consistent and the dropin on initial add should only show one.

**Comment by crookedneighbor** [comment]
The tricky thing with this is that Drop-in _does_ not actually know the full card number of the vaulted cards. In fact, it doesn't know the full card number of the newly entered card. So it's not possible for Drop-in to know if a vaulted card it's displaying and a newly added card are the same card.

We could have Drop-in fetch the customer's saved payment methods afresh every time a new card is added but that would:
* add latency to the flow, since you need to refresh the payment methods every time
* becomes tricky when drop-in is configured to display the vaulted payment methods, but allow the customer to _not_ vault a newly added credit card
* be (IMO) a confusing experience for a customer to enter a card and then not see an additional payment method get added to drop-in from the initial load

I do think the API should be updated to allow opting out of the de-duplication effect.
